ui/packages/catalog-ui-search/src/main/webapp/component/relative-time/serial.js

```diff
@@ -0,0 +1,32 @@
+/* loosely based on ISO 8601 time intervals */
+const serialize = time => {
+  if (!time || time.unit === undefined || time.last === undefined) {
+    return
+  }
+  const prefix = time.unit === 'm' || time.unit === 'h' ? 'PT' : 'P'
+  return `RELATIVE(${prefix + time.last + time.unit.toUpperCase()})`
+}
+const deserialize = value => {
+  if (!value) {
+    return
+  }
+
+  const match = value.match(/RELATIVE\(Z?([A-Z]*)(\d+\.*\d*)(.)\)/)
```

Comment:
✏️ Consider using [A-Z] instead of . for the last capture group since we're expecting an uppercase letter.